### ISSUE TITLE
Support STM32F030C8T6

### ISSUE THREAD

**Issue by slmarcos** [FastLED/FastLED#750]
Hello, anyone know to let me know about this library is compatible with the STM32F030C8T6? If it is not possible to adapt it?

**Comment by AndKe** [comment]
It is supposed to be compatible, but there is some "define" mess and pins are not correctly selectable.
Feel free to hip in on the bounty I promised in https://github.com/FastLED/FastLED/issues/744
- then somebody will fix it faster...

**Comment by AndKe** [comment]
Duplicate of #744

**Comment by zackees** [comment]
Tracked in FastLED/fbuild#385 for STM32 / STM32duino fbuild platform/build support validation.

That fbuild issue is scoped to board metadata, toolchain/backend support, and FastLED smoke-build evidence. Any remaining FastLED source/platform-code blocker should stay linked from both sides.

**Comment by zackees** [comment]
Triage update from fbuild side, tracked under FastLED/fbuild#385.

STM32F030C8T6 (Cortex-M0, STM32F0 family) is an **fbuild-side gap**. fbuild's STM32 build orchestrator currently only ships MCU configs for STM32F1, STM32F4, STM32H7, and STM32U5 - the F0 (Cortex-M0) family has no `stm32f0.json` mcu_config yet, so `fbuild build ... -e nucleo_f030r8` fails with:

```
build error: config error: unsupported STM32 MCU: 'stm32f030r8t6' (supported prefixes: stm32f103, stm32f4, stm32h7, stm32u5/stm32u585; gaps tracked by FastLED/fbuild#385: stm32f0*, stm32l4*)
```

A smoke fixture (`tests/platform/nucleo_f030r8`) has been added as a regression marker so this is unblocked once the F0 config lands. See PR FastLED/fbuild#396 and tracker FastLED/fbuild#385.

Once fbuild ships F0 support, the FastLED-side work to wire `STM32F030xx` into `src/platforms/arm/stm32/` is still required separately.

**Comment by zackees** [comment]
Closing — @AndKe already marked this a **duplicate of #744** in 2019, and #744 is closed. Recording the current STM32F0 status so the answer isn't just "duplicate".

**Short answer: not supported today, and the blocker is in the build tool rather than in FastLED's source.**

fbuild's STM32 orchestrator ships MCU configs for **STM32F1, F4, H7 and U5 only**. The F0 family (Cortex-M0) has none, so a build fails before FastLED is even reached:

```
config error: unsupported STM32 MCU: 'stm32f030r8t6'
(supported prefixes: stm32f103, stm32f4, stm32h7, stm32u5/stm32u585;
 gaps tracked by FastLED/fbuild#385: stm32f0*, stm32l4*)
```

A smoke fixture (`tests/platform/nucleo_f030r8`) exists as a regression marker, so this unblocks automatically once an F0 config lands. Tracked at **FastLED/fbuild#385**.

**And there's a second step after that**, which is easy to overlook: wiring `STM32F030xx` into `src/platforms/arm/stm32/` is separate FastLED-side work. Two things worth knowing before anyone attempts it:

- The **STM32F030C8T6 is Cortex-M0**, not M3/M4. FastLED's STM32 clockless driver goes through the M0 path (`src/platforms/arm/common/m0clockless.h`), which is cycle-counted inline assembly — so timing has to be validated on real silicon, not just compiled.
- The part has **64 KB flash / 8 KB RAM**. That's workable but tight; at 3 bytes per LED you'll want to keep strips modest and watch for the same RAM pressure that bites on ATtiny-class targets.

@AndKe's 2019 observation — *"there is some define mess and pins are not correctly selectable"* — is the FastLED-side half of this, and it's what the `src/platforms/arm/stm32/` work would address.

If you want STM32F0, follow **FastLED/fbuild#385** and open a fresh FastLED issue for the platform wiring once that lands. Closing here since the headline question is answered and the tracking already lives in two better places.
